### DISCUSSION TITLE
CIDR range/IP range/subnet Source for HTTP Client Source (or others)

### DISCUSSION THREAD

**Discussion by average-gary** [vectordotdev/vector#21277] [General]
Currently, you can only specify one IP address for an Http Client source. It would be advantageous to be able to "fan out" and make requests to multiple endpoints. 

I think you would also want to be able to specify concurrency parameters of IPs to hit per second, timeouts, and others.

From Discord:
```
jszwedko — Yesterday at 3:30 PM
Accepting multiple endpoints, I think, would be since there is precedent with some other sources like prometheus_scrape. Accepting an IP range would be a feature without precedent and so may require some more discussion, but you could certainly open a feature request to kick that discussion off
```

**Comment by EthnTuttle** [comment]
![image](https://github.com/user-attachments/assets/48bc6f72-6f45-4a73-aebe-9033fe7a1f8a)
Looks like someone considered this in the code but it is not implemented (or it is and just not documented)

**Comment by average-gary** [comment]
https://github.com/average-gary/vector/tree/destructured-http

Branch with work toward this idea. Definitely not the right implementation by any means but I'll keep tinkering. (needed something to address my needs first :) )
